Shared/mods/deathmatch/logic/CDebugHookManager.cpp

```diff
@@ -659,7 +659,9 @@ bool CDebugHookManager::CallHook(const char* szName, const std::vector<SDebugHoo
             CLuaArgument* returnedValue = *returnValues.begin();
             if (returnedValue->GetType() == LUA_TSTRING)
             {
-                if (returnedValue->GetString() == "skip")
+                // We don't want to skip the creation of new debug hooks
+                SString& strName = SString("%s", szName);
+                if (returnedValue->GetString() == "skip" && strName != "addDebugHook")
```

Comment:
use `std::string` instead:
```cpp
std::string name = szName;
if (returnedValue->GetString() == "skip" && name != "addDebugHook")
```
Or for the best solution:
```cpp
if (returnedValue->GetString() == "skip" && strcmp(name, "addDebugHook"))
```

Reply:
string_view